cdap-app-templates/cdap-etl/cdap-data-streams/src/test/java/co/cask/cdap/datastreams/DataStreamsSparkSinkTest.java

```diff
@@ -134,5 +135,6 @@ public Boolean call() throws Exception {
 
     sparkManager.stop();
     sparkManager.waitForStatus(false, 10, 1);
+    sparkManager.waitForRun(ProgramRunStatus.KILLED, 10, TimeUnit.SECONDS);
```

Comment:
They don't conflict. The issue is that the next test that runs will try to write KILLED to the store because it didn't get to finish at the end of this test, and throw an error.

This issue occurs sometimes, so its better to ensure that the program is truly killed by making sure that the run record has been written.